src/test/java/com/urbanairship/api/templates/TemplateBadRequestTest.java

```diff
@@ -0,0 +1,41 @@
+package com.urbanairship.api.templates;
+
+import com.urbanairship.api.client.ClientException;
+import com.urbanairship.api.client.Response;
+import com.urbanairship.api.client.UrbanAirshipClient;
+import com.urbanairship.api.push.model.notification.Notification;
+import com.urbanairship.api.templates.model.PartialPushPayload;
+import com.urbanairship.api.templates.model.TemplateResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class TemplateBadRequestTest {
+
+    @Test
+    public void testErrorResponseParsing() throws IOException {
+        UrbanAirshipClient client = UrbanAirshipClient.newBuilder()
+                .setKey("ISex_TTJRuarzs9-o_Gkhg")
+                .setSecret("nDq-bQ3CT92PqCIXNtQyCQ")
+                .build();
+
+        PartialPushPayload partialPushPayload = PartialPushPayload.newBuilder()
+                .setNotification(Notification.newBuilder()
+                        .setAlert("Hello {{FIRST_NAME}} {{LAST_NAME}}, this is a test!")
+                        .build()
+                )
+                .build();
+
+        TemplateRequest request = TemplateRequest.newRequest("template-id-123")
+                .setPush(partialPushPayload);
+
+        try {
+            Response<TemplateResponse> response = client.execute(request);
+        } catch (RuntimeException e) {
+            ClientException ex = (ClientException) e.getCause().getCause();
```

Comment:
I'd have an "expects exception" rule in your test, in case the exception isn't thrown.

Reply:
Ah good call, don't want a false positive